### PR TITLE
Topic: Generating & using RSA keys for JWT authentication

### PR DESCRIPTION
**Pull request by SteliosPapamichail**
This PR adds a new topic on the generation and usage of RSA keys for JWT authentication in a Ktor project. The new documentation covers the following topics:

1.  RSA Private key generation
2. Derivation of the public key from a private key
3. Extracting the exponent and modulus attributes from the public key 
4. Converting the aforementioned `e` & `n`  attributes from a HEX representation to a decimal or binary one and encoding them to Base64URL format.
5. Using these attributes in the project's `jwks.json` file and lastly,
6. Adding the private key to the project's configuration file safely by using environment variables.

This PR was inspired by my own personal issues with using RS256 along with the JWT authentication plugin in my Ktor project. The [documentation on the website](https://ktor.io/docs/jwt.html) does not provide any information on the topic of the keys, and neither does the [sample project](https://github.com/ktorio/ktor-documentation/blob/1.6.8/codeSnippets/snippets/auth-jwt-rs256/), so I decided to help others facing similar issues by describing a working process that will save them time and effort.

### CODE REVIEW

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:15]: > step first step

Typo?

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:25]: With the current command, you'll be generating an RSA key of the default key size. I suggest you add `-pkeyopt` and specifying `rsa_keygen_bits:numbits` to e.g. `2048` explicitly.

Here's the manual reference: https://www.openssl.org/docs/man3.0/man1/openssl-genpkey.html

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:56]: Do you use the terminal output of `tee` somewhere? It seems you just use the file from now on, right?

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:112]: There's a mixed use of "You" and "We". I suggest you pick one, probably the one used in the other `.md` files, for consistency.

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:186]: I don't think it's clear to everyone what this "leading 00 byte" is here. Perhaps you can explain it a bit more?

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:191]: It's safer in markdown to have an empty line before and after a section (`\n# Title\n\n`)

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:223]: It seems that lines of this file are wrapped at arbitrary points. I suggest you either wrap at a constant point, or not at all, and ideally be consistent with the other `.md` files here.

File: topics/RSA-Keys-Generation.md
Comment (daknob) [inline topics/RSA-Keys-Generation.md:238]: I suggest to change this to:

consider using environment variables or a secret store for sensitive data.

**Review by daknob — COMMENTED** [review]
Thanks for the contribution Stelios!

I must admit that I am not familiar with the Ktor project, or Kotlin for that matter, however I just thought I'd add a few comments based on my cryptography and OpenSSL experience.

Please take everything with a grain of salt, and wait for the maintainer feedback before making any large changes :)

I'm happy to review any fixes you make as a result of this review, and I'll be available to answer any questions or add any comments to this.

Again, thanks for the doc!

File: ktor.tree
Comment (SteliosPapamichail) [reply ktor.tree]: That's a great observation, fixed!

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: Thank you very much, just corrected it :)

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: Done!

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: Indeed, I missed this one, thank you!

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: That's very helpful! I improved the command so that it uses the `-pkeyopt` flag and added a reference to the manual of the command.

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: Good catch! Should be fixed along with references to "our".

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: I added an explanation based on what I could find online with a quick read. If it's not correct or enough, let me know.

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: Done :)

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: Done!

File: topics/RSA-Keys-Generation.md
Comment (SteliosPapamichail) [reply topics/RSA-Keys-Generation.md]: I'm not sure how to fix this universally for the file to be honest. Should I fiddle around with the IDEs settings for wrapping? From a quick look, other files also seem to contain such arbitrary line wrappings. I tried to improve a few that really stood out to me and I am always looking forward to fix even more :)

**Comment by SteliosPapamichail** [comment]
> Thanks for the contribution Stelios!
> 
> I must admit that I am not familiar with the Ktor project, or Kotlin for that matter, however I just thought I'd add a few comments based on my cryptography and OpenSSL experience.
> 
> Please take everything with a grain of salt, and wait for the maintainer feedback before making any large changes :)
> 
> I'm happy to review any fixes you make as a result of this review, and I'll be available to answer any questions or add any comments to this.
> 
> Again, thanks for the doc!

Thank you so much for taking the time to add such useful and actionable feedback. I've added a new commit that hopefully resolves most of these issues! Let me know if there is anything else I can fix or improve!

**Comment by daknob** [comment]
Thanks for addressing all the feedback! Regarding your question about wrapping lines, this is usually an IDE option.

For `vim`, you select a paragraph, and then you press `g` and then `q`. It's not the best if you're formatting Markdown, so you need to be a bit careful, but it generally works.

If you're writing the file, you can use `:set textwidth=79` and it will work slightly better.

**Review by vnikolova — CHANGES_REQUESTED** [review]
Awesome work, @SteliosPapamichail! Thank you for contributing to Ktor Docs. Just a few more comments from my side :)

File: topics/rsa-keys-generation.md
Comment (vnikolova) [inline topics/rsa-keys-generation.md:236]: What is a secret store? Can you please provide a relevant link with more information?

Comment (daknob) [reply topics/rsa-keys-generation.md:236]: I suggest this: https://cheatsheetseries.owasp.org/cheatsheets/Secrets_Management_Cheat_Sheet.html

File: topics/rsa-keys-generation.md
Comment (SteliosPapamichail) [reply topics/rsa-keys-generation.md]: Sounds great, fixed!

File: topics/rsa-keys-generation.md
Comment (SteliosPapamichail) [reply topics/rsa-keys-generation.md]: Perfect, thank you both for the input :)

File: topics/rsa-keys-generation.md
Comment (SteliosPapamichail) [reply topics/rsa-keys-generation.md]: Done!

File: topics/rsa-keys-generation.md
Comment (SteliosPapamichail) [reply topics/rsa-keys-generation.md]: Added a wiki link with more info on elliptic curve algorithms!

File: topics/rsa-keys-generation.md
Comment (SteliosPapamichail) [reply topics/rsa-keys-generation.md]: Amazing, didn't think of using variables for this, thank you very much :)

File: topics/rsa-keys-generation.md
Comment (SteliosPapamichail) [reply topics/rsa-keys-generation.md]: You're right, done!

**Comment by SteliosPapamichail** [comment]
> Awesome work, @SteliosPapamichail! Thank you for contributing to Ktor Docs. Just a few more comments from my side :)

Thank you! I think all changes have been implemented :) Let me know if anything else needs some work!

**Comment by vnikolova** [comment]
Thanks again @SteliosPapamichail! PR merged :raised_hands:

**Pull request merged**
